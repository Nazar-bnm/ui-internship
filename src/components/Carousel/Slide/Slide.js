import React from 'react';

import PropTypes from 'prop-types';
import './Slide.scss';

import cx from 'classnames';

export const CN = 'slide';

const Slide = ({ image, name, category }) => (
  <div className={cx(CN)}>
    <img
      className={`${CN}__image`}
      src={image}
      alt={name}
    />
    <h3 className={`${CN}__product-name title-sections`}>{name}</h3>
    <p className={`${CN}__category`}>{category}</p>
  </div>
);

Slide.propTypes = {
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  category: PropTypes.string.isRequired
};

export default Slide;
