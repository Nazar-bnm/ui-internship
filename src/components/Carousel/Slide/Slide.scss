@import "../../scss/global_styles";

.slide {
  display: inline-block;
  text-align: center;
  vertical-align: top;
  width: 200px;
  padding: 5px;
  white-space: normal;
  color: $black;

  &__image {
    width: 100%;
  }

  &__product-name{
    @include title-sections;
    color: $black;
    text-align: center;
    margin: 10px 0;
    font-size: 0.7rem;
  }

  &__category {
    @include text-sections;
    font-size: 0.5rem;
    margin: 10px 0;
    text-transform: uppercase;
  }
}

