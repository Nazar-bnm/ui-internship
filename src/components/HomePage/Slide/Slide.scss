@import '../../../scss/global_styles';

.slide-info {
    &-title {
      text-transform: uppercase; 
      font: 0.8rem font(primary-light);
    }
  
    &-description {
      margin: 0.5rem 0 1.5rem; 
      font: 2.5rem font(secondary);
      line-height: 1;
      font-variant: small-caps;
      text-transform: capitalize;
    }
  
    &-button {
      border: none;
      border-radius: 2px;
      padding: 10px 15px;
      font: 0.7rem inherit;
      text-transform: uppercase; 
      background: $black;
      color: $white;
    }
  }