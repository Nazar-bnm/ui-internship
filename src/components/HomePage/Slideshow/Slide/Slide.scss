@import '../../../../scss/global_styles';

.slide {
  &-container {
    width: 100vw;
    height: 100%;    
  }

  &-wrapper {
    @include flex($direction: column, $justify: center, $align: center);
    width: 100%;
    height: 100%;
    font-family: font(primary-light);
    background-repeat: no-repeat;
    background-size: cover;
    background-position: center center;

    &-title {
      text-transform: uppercase; 
      font: 0.8rem inherit;
    }
  
    &-description {
      margin: 0.5rem 0 1.5rem; 
      font: 2.5rem font(secondary);
      line-height: 1;
      font-variant: small-caps;
      text-transform: capitalize;
    }

    &-button {
      border: none;
      padding: 5px 10px;
      font: 0.7rem inherit;
      text-transform: uppercase; 
      background: $black;
      color: $white;
    }
  }
}

.fade {
  animation-name: fade;
  animation-duration: 3s;
  opacity: 1;
}

.zoom-in {
  animation: 2s linear  forwards zoom-in;
  opacity: 1;
  transform: scale(1);

}

.zoom-out {
  animation: 2s linear  forwards zoom-out;
  opacity: 1;
  transform: scale(1);

}

@keyframes fade {
  from {opacity: 0}
  to {opacity: 1}
}

@keyframes zoom-in {
  0% {transform: scale(1); opacity: 1;}
  100% {transform: scale(0.1); opacity: 0 ;}
}


@keyframes zoom-out {
  0% {transform: scale(1); opacity: 1;}
  100% {transform: scale(1.5); opacity: 0 ;}
}

