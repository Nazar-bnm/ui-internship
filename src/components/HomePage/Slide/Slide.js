import React from 'react';
import './Slide.scss';

const Slide = () => {
  return (<div>
    <h1 className="slide-info-title ">sales</h1>
    <p className="slide-info-description">want a new look ?</p>
    <button className="slide-info-button">
          start shopping now
    </button>
  </div>);
};


export default Slide;
