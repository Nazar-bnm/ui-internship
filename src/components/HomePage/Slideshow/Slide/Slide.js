import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import './Slide.scss';

const slide = ({ bgImage, animation, onAnimationEnd, style, children }) => {
  const bgStyle = {
    backgroundImage: `url(${bgImage})`,
  };

  const CN = 'slide';

  return (
    <div className={`${CN}-container`} style={style}>
      <div className={cx(`${CN}-wrapper`, [animation])} style = {bgStyle} onAnimationEnd={onAnimationEnd}>
        {children}
      </div>
    </div>
  );
};

slide.propTypes = {
  bgImage: PropTypes.string,
  animation: PropTypes.string,
  onAnimationEnd: PropTypes.func,
  style: PropTypes.object,
  children: PropTypes.node,
};

slide.defaultProps = {
  bgImage: '',
  animation: '',
  onAnimationEnd: () => {},
  style: {},
  children: PropTypes.node,
};

export default slide;
