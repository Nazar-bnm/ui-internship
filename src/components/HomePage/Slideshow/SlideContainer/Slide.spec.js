import React from 'react';
import { shallow } from 'enzyme';
import Slide from './Slide';

describe('<Slide />', () => {
  const CustomComponent = () => <div>Component</div>;
  it('should make a snapshot of Slide component', () => {
    const wrapper = shallow(
      <Slide bgImage="some image url" animation="some animation name" onAnimationEnd={() => {}}>
        <CustomComponent />
      </Slide>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke animation end handler when animation ends', () => {
    const mockedAnimationEndHandler = jest.fn(() => {});
    const wrapper = shallow(<Slide onAnimationEnd={mockedAnimationEndHandler} />);
    wrapper.find('.slide-wrapper').first().simulate('animationend');
    expect(mockedAnimationEndHandler.mock.calls.length).toBe(1);
  });
});
