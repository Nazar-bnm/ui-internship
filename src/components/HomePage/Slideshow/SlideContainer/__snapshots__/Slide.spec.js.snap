// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slide /> should make a snapshot of Slide component 1`] = `
<div
  className="slide-container"
  style={Object {}}
>
  <div
    className="slide-wrapper some animation name"
    onAnimationEnd={[Function]}
    style={
      Object {
        "backgroundImage": "url(some image url)",
      }
    }
  >
    <CustomComponent />
  </div>
</div>
`;
